lib/ssh/src/ssh_sftpd.erl

```diff
@@ -676,12 +676,7 @@ do_open(ReqId, State0, Path, Flags) ->
 	    ssh_xfer:xf_send_status(State0#state.xf, ReqId,
     				    ?SSH_FX_FAILURE, "File is a directory");
 	false ->
-	    AbsPath = case Root of
-			  "" ->
-			      Path;
-			  _ ->
-			      relate_file_name(Path, State0)  
-		      end,
+	    AbsPath = relate_file_name(Path, State0),
```

Comment:
I think this `AbsPath = relate_file_name(Path, State0)` should be before line 670 (`{IsDir, _FS1} = FileMod:is_dir(Path, FS0)`) above.

If `handle_op(?SSH_FXP_OPEN,...` is called (like in the test cases), the Path from the client will be used for the `FileMod:is_dir` test without any conversion to the server's file tree so the wrong Path is checked on line 670.